gnomad_qc/v4/annotations/vrs_annotation_batch_v4.py

```diff
@@ -0,0 +1,523 @@
+                " 1000 Gb"
+            )
+
+        j.storage(f"{disk_size}Gi")
+
+    j.command(
+        "set -euxo pipefail"
+    )  # set bash options for easier debugging and to make command execution more robust
+
+    return j
+
+
+def init_job_with_gcloud(
+    batch,
+    name: str = None,
+    image: str = None,
+    cpu: float = None,
+    memory: float = None,
+    disk_size: float = None,
+    mount: str = None,
+):
+    """
+    Create job and initialize glcoud authentication and gsutil commands.
+
+    Wraps Ben Weisburd's init_job (https://github.com/broadinstitute/tgg_methods/blob/master/tgg/batch/batch_utils.py#L160) with additional gcloud steps.
+    Parameters passed through to init_job:
+        :param batch: Batch object.
+        :param name: Job label which will show up in the Batch web UI.
+        :param image: Docker image name (eg. "us-central1-docker.pkg.dev/broad-mpg-gnomad/ga4gh-vrs/marten_0615_vrs0_8_4").
+        :param cpu: Number of CPUs (between 0.25 to 16).
+        :param memory: Amount of RAM in Gb (eg. 3.75).
+        :param disk_size: Amount of disk in Gb (eg. 50).
+        :param mount: Name of GCP Bucket to mount using cloudfuse.
+        :return: New job object.
+    """
+    job = init_job(batch, name, image, cpu, memory, disk_size)
+    # Retry gcloud auth and gsutil commands to avoid transient failures
+    job.command(
+        """
+    retry() {
+              "$@" ||
+                  (sleep 2 && "$@") ||
+                  (sleep 5 && "$@");
+    }
+    retry gcloud -q auth activate-service-account --key-file=/gsa-key/key.json
+
+    curl_and_python() {
+        curl -sSL broad.io/install-gcs-connector | python3
+    }
+    retry curl_and_python
+    """
+    )
+    if mount:
+        job.cloudfuse(mount, "/local-vrs-mount")
+    return job
+
+
+def main(args):
+    """Generate VRS annotations for a Hail Table of variants."""
+    # Initialize Hail w/chosen mode (spark or batch for QoB), setting global
+    # seed for if user decides to downsample
+    hl.init(
+        backend=args.backend_mode,
+        tmp_dir=args.tmp_dir_hail,
+        gcs_requester_pays_configuration="broad-mpg-gnomad",
+        default_reference="GRCh38",
+        global_seed=args.hail_rand_seed,
+        regions=["us-central1"],
+    )
+
+    logger.info(f"Hail version as: {hl.version()}")
+
+    # Example schema
+    logger.info("Example VRS schema for 2 variants:")
+    ht_example = hl.read_table("gs://gnomad-vrs-io-finals/ht-inputs/two-snps-schema.ht")
+    ht_example.show()
+
+    working_bucket = args.working_bucket
+    version = args.version
+
+    # Prefix to create custom named versions of outputs
+    prefix = args.prefix + version
+
+    input_paths_dict = {
+        "v4.0_exomes": f"gs://gnomad-qin/v4_annotations/v4_vds_all_variants.ht",
+        "test_v4.0_exomes": f"gs://gnomad-qin/v4_annotations/v4_vds_2_partitions.ht",
+        "v4.0_supp1": f"gs://gnomad-qin/v4_annotations/v4_vds_supp1.ht",
```

Comment:
can we move these inputs to gnomad buckets rather than personal buckets? the test files can go in "gs://gnomad-vrs-io-finals" under a v4 directory. And the actual input for the final run ideally would come from a gnomad resource

Reply:
I thought these are intermediate files and they will be deleted. Do we need to keep them? It's basically just locus and alleles.

Reply:
was there any discussion in a meeting regarding what the input ht should be? the main input should be something that's easy to locate and run on again

Reply:
I'm using the VEP output as input for now, but we can change it later if we need to rerun on release HT.